File: 240409/1이 되는 순간까지/until-the-moment-I-reach-one.py
N = int(input())

def recur_eval_sum(n):
    if n == 1:
        return 0

    if n % 2 == 0:
        return recur_eval_sum(n // 2) + 1
    else:
        return recur_eval_sum(n // 3) + 1

print(recur_eval_sum(N))

File: 251226/1이 되는 순간까지/until-the-moment-I-reach-one.cpp
#include <iostream>

using namespace std;

int N;
int recur_eval_sum(int num);

int main() 
{
    cin >> N;

    // Please write your code here.
    cout << recur_eval_sum(N);
    return 0;
}

int recur_eval_sum(int num)
{
    if(num == 1)
        return 0;

    if(num % 2 == 0)
        return recur_eval_sum((int)(num / 2)) + 1;
    else
        return recur_eval_sum((int)(num / 3)) + 1;
}